File: src/python/problem_009.py
def is_pythagorean_triplet(a, b, c):
    return a ** 2 + b ** 2 == c ** 2

def euler_009():
    n_sum = 1000
    for a in range(1, n_sum // 3):
        for b in range(a + 1, (n_sum - a) // 2):
            c = n_sum - a - b
            if is_pythagorean_triplet(a, b, c):
                print(a, b, c, a * b * c)

if __name__ == '__main__':
    euler_009()

File: src/go/problem_009.go
package main

import (
	"fmt"
)

func isPythagoreanTriplet(a, b, c int) bool {
	return a * a + b * b == c * c
}

func main() {
	target := 1000

	for a := 1; a < target / 3; a++ {
		for b := a + 1; b < target / 2; b++ {
			c := target - a - b
			if isPythagoreanTriplet(a, b, c) {
				fmt.Println(a, b, c)
				fmt.Println(a * b * c)
				return
			}
		}
	}
}
